Add Product Collection Full Grid pattern

kmanijak opened the pull request:
<!-- Please do not remove any information from this pull request. Instead, add N/A or leave blank if not applicable -->

## What

Add Product Collection Full Grid pattern.

Additionally, when implementing I realized the Product Button doesn't react on the "Text align" property. Be mindful in this context the initial purpose was to align the whole button (as per https://github.com/woocommerce/woocommerce-blocks/pull/7816) which was preserved in this PR.

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/9369

## Why

This is one of the patterns from a set of patterns we'd like to provide for Product Collection

## Testing Instructions

<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

0. At this moment Product Collection is still experimental feature (PR to make it Core feature: https://github.com/woocommerce/woocommerce-blocks/pull/10524), hence the pattern is not included in the inserter. In order to test the pattern please remove this line: https://github.com/woocommerce/woocommerce-blocks/pull/10660/files#diff-d26dedf0a7efa88eae30cbe732192f2769376839845875e6e57505d33f19653cR6.
1. Go to Editor
3. Insert "Product Collection Full Grid" pattern
4. Save and go to frontend
5. Verify it looks like in the screenshots below

Known limitations:
- Product Price cannot be styled according to designs. This work is being covered in https://github.com/woocommerce/woocommerce-blocks/pull/9398

* [ ] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested (ie: it makes changes to tests, coding standards, docblocks, etc.). -->
* [ ] Should be tested by the development team exclusively <!-- Check this box if this PR should be tested by the development team exclusively (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

## Screenshots or screencast

<!-- Any screenshots of UI changes will be helpful to include here. Leave blank if not applicable. -->

Reference | Editor | Frontend |
| ------ | ----- | ---|
| ![image](https://github.com/woocommerce/woocommerce-blocks/assets/20098064/b8efebdb-dbfc-4181-93bf-cc0bcd8c144c) |    <img width="802" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/20098064/f3d37fe3-e1f0-4a85-8894-2bcd98f6d46f">   |    <img width="702" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/20098064/3a3f35c7-9ef0-4eab-96b7-1bb834ae7ad6">   |

## WooCommerce Visibility

<!-- Check this documentation link (../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WooCommerce core, part of the feature plugin, or experimental. -->
Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:
* [x] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog
<!-- Provide a brief, descriptive summary of the changes in this PR. Include potential impacts on different parts of the product. Example: "Updated the checkout process to streamline the experience for users and reduce the number of steps." -->

> Product Collection: new pattern "Product Collection Full Grid"

kmanijak committed:
Add initial look of Full Grid Product Collection pattern

kmanijak committed:
Refactor textAlign property of Product Button so it uses flex rather than text-align

kmanijak committed:
Update Product Button text align after the fix

kmanijak committed:
Merge branch 'trunk' into add/9369-product-collection-full-grid-pattern

kmanijak committed:
Remove debug log (#10719)

kmanijak committed:
Add Product Collection Grid pattern (#10660)

opr committed:
Dispatch the `wc-blocks_render_blocks_frontend` event when rendering the empty cart block (#10619)

dinhtungdu committed:
E2E: Refactor Mini Cart to be ready for fully parallel (#10704)

dinhtungdu reviewed (commented):
Thanks for the PR @kmanijak! I have a question and an issue testing this PR:

- From the visual look, the product button is what differs this from #10660, right? I was confused by the naming. Initially, I thought Full Grid refers to the grid pattern with full width, it turned out a grid with more elements.
- The button alignment didn't work on my test, it's always center regardless of setting. Both my local build and the zip file built for this PR gave me the same result:

<img width="703" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/5423135/12154d3a-3c3a-440f-b9db-c6953baf0066">

kmanijak committed:
Revert dequeue add-to-cart-variation script which is needed to properly handle variable products in single product page (#10723)

kmanijak committed:
Update alignment options to new setting

kmanijak commented:
> From the visual look, the product button is what differs this from https://github.com/woocommerce/woocommerce-blocks/pull/10660, right? I was confused by the naming. Initially, I thought Full Grid refers to the grid pattern with full width, it turned out a grid with more elements.

I'm following the designs and naming convention from pdnLyh-3uF-p2.

> The button alignment didn't work on my test, it's always center regardless of setting. Both my local build and the zip file built for this PR gave me the same result:

It's funny, after preparing this PR and before you reviewed I changed the wrapper's `flex-direction` to column (https://github.com/woocommerce/woocommerce-blocks/commit/6ed0eb11fba8d8f488617dc1e9dd92aa5b828b45) and rebased this branch. So when you tested, the `justify-content` property didn't work anymore. I changed that to `align-items` in https://github.com/woocommerce/woocommerce-blocks/pull/10665/commits/00f1e0d2c12f54cf5752cb4757f22a6691f66119 and now it should work correctly. 

@dinhtungdu can I ask you for re-review? 🙌

dinhtungdu committed:
Merge branch 'trunk' into add/9369-product-collection-full-grid-pattern

dinhtungdu reviewed (approved):
Thanks for the update! This is LGTM now!
